### PR TITLE
added exception to handle negative or 0 current price. Added ability to run multiple invalid inputs into instrument type

### PR DESCRIPTION
**Pull request by williamlamjy**
… exception handling non-number inputs for price
added valid instrument chechks in validate

links #186

### CODE REVIEW

File: src/main/java/seedu/mtracker/console/InputParser.java [hunk 1]
```diff
@@ -56,6 +56,9 @@ public int getInstrumentNumber() {
     public AddInstrumentCommand getAddInstrumentParameters() throws InvalidInstrumentError {
         TextUi.displayAddInstrumentFirstInstruction();
         String addInstrumentType = getUserInput();
+        while (!Validate.isValidInstrument(addInstrumentType)) {
+            addInstrumentType = getUserInput();
+        }
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/console/InputParser.java:61]: Use do while loop here

**Review by theodorekwok — COMMENTED** [review]
Nice work, just one comment

File: src/main/java/seedu/mtracker/commons/Validate.java [hunk 2]
```diff
@@ -47,6 +53,35 @@ public static void checkName(String name, String instrumentType) throws InvalidN
         }
     }
 
+    public static boolean isInvalidInstrument(String instrument) {
+        switch (instrument) {
+        case AddStockCommand.COMMAND_WORD:
+        case AddCryptoCommand.COMMAND_WORD:
+        case AddForexCommand.COMMAND_WORD:
+        case AddEtfCommand.COMMAND_WORD:
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/commons/Validate.java:61]: Sry missed this the first time
```suggestion
        case AddStockCommand.COMMAND_WORD:
              // Fallthrough
        case AddCryptoCommand.COMMAND_WORD:
              // Fallthrough
        case AddForexCommand.COMMAND_WORD:
              // Fallthrough
        case AddEtfCommand.COMMAND_WORD:
```

according to the java coding standards

File: src/main/java/seedu/mtracker/commons/Validate.java [hunk 3]
```diff
@@ -47,6 +53,35 @@ public static void checkName(String name, String instrumentType) throws InvalidN
...
+            return false;
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/commons/Validate.java:62]: Add break statement here. If its better to use an if else feel free to do so

**Review by theodorekwok — COMMENTED** [review]
Sry just 2 minor comments

File: src/main/java/seedu/mtracker/commons/Validate.java (hunk 3 above)
Comment (williamlamjy) [reply src/main/java/seedu/mtracker/commons/Validate.java:62]: adding break here leads to an error of unreachable statement for some reason

Comment (theodorekwok) [reply src/main/java/seedu/mtracker/commons/Validate.java:66]: Ohh its cause u are returning, since that's the case I think its ok

**Review by theodorekwok — APPROVED** [review]
LGTM!

**Pull request merged**